chore: updated ubuntu version pinning and some minor errors related to deployment

JagjeevanAK opened the pull request:
Signed-off-by: Jagjeevan [83103469+JagjeevanAK@users.noreply.github.com](mailto:83103469+JagjeevanAK@users.noreply.github.com)

### What
Updated ubuntu version pinning to latest and some minor errors related to deployment

### Screenshot
<img width="1343" alt="Screenshot 2025-05-05 at 6 45 47 AM" src="https://github.com/user-attachments/assets/6edd791d-9463-49a8-b090-2dc7be0cb5d6" />

### Fixes bug(s)
#1557

JagjeevanAK commented:
now deployment should be smooth as-usual.

alexgarel reviewed (approved):
Thank you, it's almost good, but it's important to pin down version number.

alexgarel reviewed (changes requested):
Hum we should check the python version on 24.04 and modify python version used in validation workflow accordingly.

alexgarel commented:
@JagjeevanAK python version in ubuntu 24.04 is 3.12, so it's better we change actions to use this version, to be sure it works.

JagjeevanAK commented on .github/workflows/generate-docs.yml:
```diff
@@ -17,7 +17,7 @@ on:
 jobs:
   publish-docs:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Okay, but then we need to update things everytime. I will look into it to make it automatically update everytime.

JagjeevanAK commented:
Hey @alexgarel, I have updated the few changes and removed the matrix strategy form the action.

but the thing is that when python version is 3.12 then things are breaking, main issue is with HDF5 lib.

error logs :

```typescript
304 | error: Unable to load dependency HDF5, make sure HDF5 is installed properly
305 | error: libhdf5.so: cannot open shared object file: No such file or directory
```

JagjeevanAK commented:
This PR is ready to merge. solve the issue by updating the h5py verison.

alexgarel reviewed (approved):
Perfect, thanks.
